src/entity_ref.rs

```diff
@@ -3,18 +3,32 @@ use core::ops::{Deref, DerefMut};
 use core::ptr::NonNull;
 
 use crate::archetype::Archetype;
-use crate::{Component, MissingComponent, Query, QueryOne};
+use crate::{Component, Entity, MissingComponent, Query, QueryOne};
 
 /// Handle to an entity with any component types
 #[derive(Copy, Clone)]
 pub struct EntityRef<'a> {
     archetype: &'a Archetype,
+    generation: u32,
```

Comment:
Since this borrows `&'a Archetype`, I guess it could also borrow '&'a [EntityMetadata]` instead of storing the generation? So resolving the generation lazily making `world::Iter::next` faster but `EnityRef::entity` slower? Not sure whether this a better trade off though and whether this is measurable at all.

Reply:
Yeah, I dunno. I think I'll favor simplicity unless someone cares enough to profile.